schema/table.go

```diff
@@ -194,47 +199,40 @@ func (t Table) Resolve(ctx context.Context, meta ClientMeta, parent *Resource, r
 			continue
 		}
 		for i := range objects {
-			resource := NewResourceData(&t, parent, objects[i])
-			if t.PreResourceResolver != nil {
-				if err := t.PreResourceResolver(ctx, meta, resource); err != nil {
-					meta.Logger().Error().Str("table", t.Name).Err(err).Msg("pre resource resolver failed")
-					atomic.AddUint64(&summary.Errors, 1)
-					continue
-				}
-
-				meta.Logger().Trace().Str("table", t.Name).Msg("pre resource resolver finished successfully")
-			}
-			columnSummary := t.resolveColumns(ctx, meta, resource)
-			atomic.AddUint64(&summary.Errors, columnSummary.Errors)
-			summary.Errors += columnSummary.Errors
-			if t.PostResourceResolver != nil {
-				meta.Logger().Trace().Str("table", t.Name).Msg("post resource resolver started")
-				if err := t.PostResourceResolver(ctx, meta, resource); err != nil {
-					meta.Logger().Error().Str("table", t.Name).Stack().Err(err).Msg("post resource resolver finished with error")
-					atomic.AddUint64(&summary.Errors, 1)
-				} else {
-					meta.Logger().Trace().Str("table", t.Name).Msg("post resource resolver finished successfully")
-				}
-			}
-
-			atomic.AddUint64(&summary.Resources, 1)
-			resolvedResources <- resource
-			for _, rel := range t.Relations {
-				relationSummary := rel.Resolve(ctx, meta, resource, resolvedResources)
-				atomic.AddUint64(&summary.Errors, relationSummary.Errors)
-				atomic.AddUint64(&summary.Panics, relationSummary.Panics)
-				atomic.AddUint64(&summary.Resources, relationSummary.Resources)
-			}
+			summary.Merge(t.resolveObject(ctx, meta, parent, objects[i], resolvedResources))
```

Comment:
Does this work with summary being non-ptr? Maybe return the new value from `Merge` and set it to summary?
`summary = summary.Merge(t.resolveObject(...))`

Reply:
I think that work because `Merge` has a pointer `receiver`